app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java

```diff
@@ -65,9 +66,13 @@
     @POST("/cgi/product_image_upload.pl")
     Call<JsonNode> saveImage(@PartMap Map<String, RequestBody> fields);
 
+    @GET("brand/{brand}.json")
+    Call<Brand> getBrand(@Path("brand") String brand);
```

Comment:
I think `byBrand` also does the same thing there is no need for this function. we can use `Search.java`  for this purpose there is no need for `Brand.java` that's almost the same.

Reply:
we can paginate the results and implement infinite scrolling same way as it is done in `searchProductByName` function, if not loading will be sluggish.

Reply:
yeah, sure I would look at it, thanx :). Also, I'm working to get other pages of the list as well.